packages/java-parser/src/productions/classes.js

```diff
@@ -606,9 +606,14 @@ function defineRules($, t) {
   // https://docs.oracle.com/javase/specs/jls/se11/html/jls-8.html#jls-EnumConstantList
   $.RULE("enumConstantList", () => {
     $.SUBRULE($.enumConstant);
-    $.MANY(() => {
-      $.CONSUME(t.Comma);
-      $.SUBRULE2($.enumConstant);
+    $.MANY({
+      GATE: () =>
+        $.LA(1).tokenType === t.Comma &&
```

Comment:
The Gate is an **additional** constraint
So the $.LA(1).tokenType === t.Comma is already checked.

It only has a very small runtime const, but maybe it is best to avoid unneeded checks